server/src/main/java/io/crate/replication/logical/action/PublicationsStateAction.java

```diff
@@ -183,13 +206,28 @@ protected ClusterBlockException checkBlock(Request request,
                         }
                         return false;
                     })
+                    .filter(t -> userCanPublish(t, user))
                     .map(RelationInfo::ident)
                     .toList();
             } else {
+                // No need to call userCanPublish() since for the pre-defined tables case this check was already done on the publication creation.
                 relationNames = publication.tables();
             }
             return relationNames;
         }
+
+        private static boolean userCanPublish(TableInfo t, User publicationOwner) {
+            for (Privilege.Type type: READ_WRITE_DEFINE) {
+                // This check is triggered only on ALL TABLES case.
+                // Required privileges correspond to those we check for the pre-defined tables case in AccessControlImpl.visitCreatePublication.
+
+                // Schemas.DOC_SCHEMA_NAME is a dummy parameter since we are passing fqn as ident.
+                if (!publicationOwner.hasPrivilege(type, Privilege.Clazz.TABLE, t.ident().fqn(), Schemas.DOC_SCHEMA_NAME)) {
```

Comment:
Normally we use `currentSession.defaultSchema` - but it's one of DOC or PG_CATALOG - and here we just iterate over all schemas and their FQN-s so I think last param doesn't matter here